cdap-common/bin/functions.sh

```diff
@@ -503,31 +503,15 @@ cdap_set_spark() {
       # If there is no valid SPARK_HOME, we should unset the existing one if it is set
       # Otherwise the spark-shell won't run correctly
       unset SPARK_HOME
-      local __spark_shell=$(which spark-shell 2>/dev/null)
```

Comment:
these changes are from https://github.com/caskdata/cdap/pull/8893... looks like this branch needs updated.